mentat/conversation.py

```diff
@@ -52,7 +55,25 @@ def get_model_response(self, config: ConfigManager) -> (str, list[CodeChange]):
         messages = self.messages.copy()
 
         code_message = self.code_file_manager.get_code_message()
-        messages.append({"role": "system", "content": code_message})
+        system_message = code_message
+
+        if self.code_map:
+            code_map_message = self.code_map.get_message()
+            if code_map_message is not None:
+                code_map_message_token_count = count_tokens(code_map_message)
+                prompt_token_count = 0
+                prompt_token_count += count_tokens(code_message)
+                for message in messages:
+                    prompt_token_count += count_tokens(message["content"])
+                token_buffer = 1000
+                token_limit = 32768 if self.allow_32k else 8192
+                token_count = (
+                    prompt_token_count + code_map_message_token_count + token_buffer
+                )
+                if token_count < token_limit:
+                    system_message = "\n".join([code_message, code_map_message])
```

Comment:
Since system_message is set to code_message up there, I think it would be a lot cleaner to do += f"\n{code_map_message}"